### ISSUE TITLE
ID 261625: [Programmatic Access - Simon - Edit Panel]: Aria label is not defined for "expand/collapse" button present for "Data source: AWS IoT Site wise".

### ISSUE THREAD

**Issue by eledobleefe** [grafana/grafana#66350]
### **User Experience:**  
Visually impaired people who depend on screen reader users will not get information about the control as aria label is not defined for the control due to which they will not be able to understand the purpose of it and cannot perform any corresponding action.
_Note: User credentials should NOT be included in the bug._

### **Repro Steps:**

1. Open #Url: [play.grafana.org](https://play.grafana.org/) Sign in with valid credentials. "Home" page appears. 
2. Navigate to the left navigation bar and to the "Alerting" button, select it. 
3. It gets expanded and navigates to the expanded sub menu items. 
4. Select the "Alert Rules" menu items. A new page "Alerting" appears. 
5. Navigate the tab " Alert Rules". Select "Data source: AWS IoT Site wise" 
6. Navigate to "Grafana" section. 
7. Expand the "Data source: AWS IoT Site wise" button and navigate all the controls present under it. 
8. Navigate the fields such as, "Labels", "Go to Dashboard" etc. controls. 
9. Run AI tool. Verify the issue.

**Actual Result:**
`Aria-label` is not defined for the "expand/collapse" button present for "Data source: AWS IoT Site wise".
Screen reader behavior: While navigating on the "expand/collapse" button screen reader (Narrator) announces only the state as "button expanded/button collapsed". As the aria label is empty.

**Expected Result:**
`Aria-label` should be defined as "Data source: AWS IoT Site wise" for "expand/collapse" button present. So that screen reader announces as Grafana "Data source: AWS IoT Site wise button expanded/button collapsed".

**Issue:**
Ensures buttons have discernible text (button-name - https://accessibilityinsights.io/info-examples/web/button-name) 

**Target application:**  
Alerting: Alert rules - Grafana - https://play.grafana.org/alerting/list?dataSource=AWS%20IoT%20SiteWise 

**Element path:** 
`.css-1ybakrp-button`

**Snippet:**  
`<button class="css-1ybakrp-button" type="button" aria-expanded="true" data-testid="group-collapse-toggle">`

How to fix:
Fix any of the following:   

- Element does not have inner text that is visible to screen readers.   
- `aria-label` attribute does not exist or is empty.   
- `aria-labelledby` attribute does not exist, references elements that do not exist or references elements that are empty.   
- Element has no `title` attribute.   
- Element's default semantics were not overridden with `role="none"` or `role="presentation"`.

**Environment:**  
Microsoft Edge Version 105.0.1300.0
This accessibility issue was found using Accessibility Insights for Web 2.32.0 (axe-core 4.4.1), a tool that helps find and fix accessibility issues.  

### **Severity:**
`SEV2`

### **Area:**
`AlertGroup Component`

**Comment by grafanabot** [comment]
@eledobleefe please add one or more appropriate labels. Here are some tips:

- if you are making an issue, TODO, or reminder for yourself or your team, please add one label that best describes the product or feature area. Please also add the issue to your project board. :rocket:

- if you are making an issue for any other reason (docs typo, you found a bug, etc), please add at least one label that best describes the product or feature that you are discussing (e.g. `area/alerting`, `datasource/loki`, `type/docs`, `type/bug`, etc). [Our issue triage](https://github.com/grafana/grafana/blob/main/ISSUE_TRIAGE.md#3-categorizing-an-issue) doc also provides additional guidance on labeling. :rocket:

 Thank you! :heart:
